bin/http-signature-header-js

```diff
@@ -0,0 +1,208 @@
+}
+
+function makeHTTPHeaders(headers = {}) {
+  let message = '';
+  for(const key in headers) {
+    let value = headers[key];
+    if(Array.isArray(value)) {
+      value = value.join(',');
+    }
+    message += `${key}: ${value}\n`;
+  }
+  return message;
+}
+
+program
+  .command('c14n')
+  .alias('cannonize')
+  .action(async () => {
+    try {
+      const {headers} = program;
+      const requestOptions = await getHTTPMessage();
+      if(!headers) {throw new Error(
+        '--headers required for command c14n|cannonize' +
+        'ex: --header date,etag');}
+      const includeHeaders = program.headers ? program.headers.split(/\s/) : '';
+      const result = httpSigs.
+        createSignatureString({includeHeaders, requestOptions});
+      console.log(result);
+      process.exit(0);
+    } catch(e) {
+      throw e;
+      process.exit(1);
+    }
+  });
+
+program
+  .command('sign')
+  .action(async () => {
+    try {
+      const {
+        headers,
+        keyId,
+        privateKey,
+        keyType,
+        algorithm,
+        created,
+        expires
+      } = program;
+      const privateKeyFile = await readFile(privateKey);
+      const includeHeaders = headers ? program.headers.split(/\s/) : '';
+      const requestOptions = await getHTTPMessage();
+      const options = {
+        keyId,
+        algorithm,
+        includeHeaders,
+        keyId,
+        requestOptions,
+        privateKey: privateKeyFile
+      };
+      /**
+       * Add HTTPSignatures headers to a given `requestOptions` object.
+       * This is compatible with both `request` and `axios` libraries.
+       *
+       * @param {string} algorithm -
+       * The signing algorithm to use (e.g. rsa-sha256, ed25519).
+       * @param {Object} identity -
+       * The identity as typically constructed in mock.data.js.
+       * @param {Object} requestOptions -
+       * The request options.
+      */
+      const createHttpSignatureRequest = async (
+        {algorithm = 'ed25519', privateKey, keyId,
+          requestOptions, includeHeaders = []}) => {
+        requestOptions.headers = requestOptions.headers || {};
+        if(!requestOptions.headers.date) {
+          requestOptions.headers.date = jsprim.rfc1123(new Date());
+        }
+        const plaintext = httpSigs.createSignatureString(
+          {includeHeaders, requestOptions});
+        const authzHeaderOptions = {includeHeaders, keyId};
+        const cryptoOptions = {plaintext};
+        if(algorithm.startsWith('rsa')) {
+          authzHeaderOptions.algorithm = algorithm;
+          const alg = algorithm.split('-');
+          cryptoOptions.algorithm = alg[0];
+          cryptoOptions.privateKeyPem = privateKey;
+          cryptoOptions.hashType = alg[1];
+        }
+        if(algorithm === 'ed25519') {
+          cryptoOptions.algorithm = algorithm;
+          cryptoOptions.privateKeyBase58 = bs58.encode(privateKey);
+        }
+        authzHeaderOptions.signature =
```

Comment:
Should this throw here if `algorithm` is not an expected type?

Reply:
it actually should error if the algorithm is not in the http signatures algorithm repository.